### DISCUSSION TITLE
How to manipulate fields in child table when fetching parent table with include keyword?

### DISCUSSION THREAD

**Discussion by leungkinghin-ct** [prisma/prisma#25437] [Q&A] [answered]
### Question

Hello,

In our system we had table A which was parent table while table B was its child table.
When fetching table A, we made use of arguments as below so that data in table B were also fetched.

```
include: {
   B: true,
},
```
Now we would like to manipulate one of fields in table B during retrieving table A, therefore we make use extended prisma client to archive it.

```
.$extends({
      query: {
        B: {
          async findMany({ args, query }) {
            return query(
              await manipulation(args)
            )
          },
          async findFirst({ args, query }) {
            return query(
              await manipulation(args)
            )
          },
        },
      },
    })
```
However, when we try to use `query` in extended client, we don't find the manipulation apply on the specific field from table B. 

Here are our questions
1. How to use `query` to manipulate fields from child table when fetching parent table with `include`?
2. If using `result` instead, how to make the `compute` work since the manipulation function is retuning Promise?

Thank you.

### Information about Prisma Schema, Client Queries and Environment (optional)

```prisma
// Add your schema.prisma
```

```ts
// Add any relevant Prisma Client queries here
```

- OS: 
- Database: 
- Node.js version: 

- Run `prisma -v` to see your Prisma version and paste it

**Comment by ludralph** [comment, marked as answer]
Hi @leungkinghin-ct 👋 

> How to use query to manipulate fields from child table when fetching parent table with include?

At the moment, `query` extensions are only called for top-level models. We do have a [feature request](https://github.com/prisma/prisma/issues/24525) for supporting nested queries in `query` extension. See also this closed [issue](https://github.com/prisma/prisma/issues/22396).

> If using result instead, how to make the compute work since the manipulation function is retuning Promise?

For this question, you may want to take a look at the comment [here](https://github.com/prisma/prisma/issues/18300#issuecomment-1468303756) and the answer [here](https://stackoverflow.com/questions/76341548/prisma-client-extensions-computed-fields-returns-promise).
